### PR TITLE
Fix Environment Config

### PR DESCRIPTION
**Pull request by Fosol**
There were a number of issues setting up new environments.  Ports are generally easier to configure/change with this update.

## Summary

- Changed all of the default ports, so you will need to run `make down` and `make nuke` to get started again
- Bash script variables were not working due to incorrect implementation
- `make nuke` will delete your configuration files.  It does make a backup of your files before deleting (the first time).
- `make init` will create configuration files and start `core` and `kafka` containers up
- `make utility-up` will start `utility` containers
- `make service-up` will start `service` containers

## ToDo

Still running into issues getting the **Dejavu** utility to connect to Elasticsearch for some reason

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -37,9 +37,9 @@ help:
 # Solution Configuration
 ##############################################################################
 
+
 setup: ## Setup local environment for development, generate configuration files.
 	$(info Setup local environment for development, generate configuration files.)
-	@./tools/scripts/variables.sh
```
Comment (Fosol) [inline Makefile:42]: Variables cannot be shared this way regrettably

File: Makefile [hunk 2]
```diff
@@ -53,25 +53,30 @@ init: ## Initialize your local environment and start the core solution.
 	@make db-update
 	@make elastic-update
 
+nuke: ## Stop all containers, delete all containers, volumes, and configuration
```
Comment (Fosol) [inline Makefile:56]: Remove everything to get started again

File: Makefile [hunk 3]
```diff
@@ -123,42 +123,52 @@ kafka-stop: ## Stops the kafka containers or the one specified (n=service name)
...
 service-down: ## Stops the service containers and removes them
 	$(info Stops the service containers and removes them)
-	@docker-compose -f docker-compose.yml -f docker-compose.override.yml -f ./db/kafka/docker-compose.yml --profile service --profile utility down
+	@docker-compose -f docker-compose.yml -f docker-compose.override.yml -f ./db/kafka/docker-compose.yml --profile service down -v
+
+##############################################################################
+# Utilities Management
+##############################################################################
+
+utility-build: ## Builds the utility containers, or the one specified (n=service name)
```
Comment (Fosol) [inline Makefile:157]: Separated utility commands

File: db/elasticsearch/elasticsearch.yml [hunk 4]
```diff
@@ -55,7 +55,7 @@ bootstrap:
 http:
   port: 9200
   cors:
-    allow-origin: http://localhost:50009,http://127.0.0.1:50009,https://opensource.appbase.io
+    allow-origin: http://localhost:40005,http://127.0.0.1:40005,https://opensource.appbase.io
```
Comment (Fosol) [inline db/elasticsearch/elasticsearch.yml:58]: This is still hardcoded, but should be switched with a variable at some point to enable future changes to ports

File: db/elasticsearch/scripts/migration.sh [hunk 5]
```diff
@@ -86,6 +90,14 @@ if [ -z "$password" ]; then
   fi
 fi
 
+if [ -z "$url" ]; then
```
Comment (Fosol) [inline db/elasticsearch/scripts/migration.sh:93]: Added new argument to configure elastic url

File: tools/scripts/gen-env-files.sh [hunk 6]
```diff
@@ -14,7 +16,64 @@ if test -f "./.env"; then
     echo "./.env exists"
 else
 echo \
-"" >> ./.env
+"#############################
+# Network
+#############################
+
+NGINX_HTTP_PORT=$portNginx
```
Comment (Fosol) [inline tools/scripts/gen-env-files.sh:23]: This should enable us to change ports easier in the future

File: tools/scripts/variables.sh [hunk 7]
```diff
@@ -1,94 +1,168 @@
 #!/bin/bash
 
-varKeycloak=$(grep -Po 'KEYCLOAK_USER=\K.*$' ./auth/keycloak/.env)
-if [ -z "$varKeycloak" ]
+######################################################################
+# Keycloak configuration
+######################################################################
+
+export keycloakDbName="keycloak"
```
Comment (Fosol) [inline tools/scripts/variables.sh:7]: Export variables so they can be used in other scripts

File: Makefile [hunk 8]
```diff
@@ -198,17 +209,17 @@ db-refresh: ## Stop and delete the database container and volume, then rebuilds
...
 ##############################################################################
-# Kafka Utility Commands
+# Kafka Commands
 ##############################################################################
 
 reset-consumer-offset: ## Reset the consumer group topic offset.
 	$(info Reset the consumer group topic offset)
 	@cd ./db/kafka/broker/scripts; ./reset-consumer-offset.sh
 
+kafka-update: ## Run the kafka migration (n=migration name, r=rollback, z=zookeeper)
```
Comment (Fosol) [inline Makefile:221]: We can now run scripts based on a version number for kafka initialization and migrations

File: db/kafka/migrations/U1.0.0.sh [hunk 9]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+# Logging topics
```
Comment (Fosol) [inline db/kafka/migrations/U1.0.0.sh:3]: Provide a rollback to V1.0.0.
Using same convention as flyway database migration

**Review by sturufous — APPROVED** [review]

File: db/kafka/migrations/V1.0.0.sh [hunk 10]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+# Logging topics
```
Comment (Fosol) [inline db/kafka/migrations/V1.0.0.sh:3]: Provide a upgrade migration to V1.0.0.
Using same convention as flyway database migration

**Pull request merged**